### PR TITLE
Gh-987: Upgrade dependencies

### LINKED ISSUE #987: Upgrade dependencies for Gaffer 2
Lots of dependencies are being upgraded in Gaffer 2: https://github.com/gchq/Gaffer/pull/2608
Ensure tools is upgraded in the same way.

Includes cherry-picking https://github.com/gchq/gaffer-tools/issues/986

### PR DESCRIPTION
**Pull request by GCHQDeveloper314**
# Related Issue

- Resolve #987

### CODE REVIEW

**Comment by GCHQDeveloper314** [comment]
This PR is for compatibility with Gaffer 2 alpha 2. CI will fail until that version is used in Gaffer tools, this branch will then build.

File: performance-testing/performance-testing-core/src/main/java/uk/gov/gchq/gaffer/performancetesting/query/QueryTest.java [hunk 1]
```diff
@@ -114,7 +113,7 @@ private void queryBatch(final Supplier<? extends ElementSeed> elementSeedSupplie
         final long startTime = System.currentTimeMillis();
         final long numResults;
         try {
-            final CloseableIterable<? extends Element> results = graph.execute(getElements, new User());
+            final Iterable<? extends Element> results = graph.execute(getElements, new User());
```
Comment (t92549) [inline performance-testing/performance-testing-core/src/main/java/uk/gov/gchq/gaffer/performancetesting/query/QueryTest.java:116]: Does this need to be closed in a `finally` block? I know it wasn't before, just will be a memory leak if there is an exception (I think).

**Comment by GCHQDeveloper314** [comment]
Because of [Gaffer #2634](https://github.com/gchq/Gaffer/issues/2634) tests which require WARs are failing. This will be fixed in the next alpha.

**Comment by GCHQDeveloper314** [comment]
This PR will need to be merged as is, unless we delay a release until alpha 3. The Gaffer WARs for alpha 2 are broken, they are used for these CI tests and so they will always fail.

The WAR problem was fixed in [Gaffer #2635](https://github.com/gchq/Gaffer/pull/2635). To test this branch, clone 2635 and build it, then clone this branch, change the gaffer versions to the SNAPSHOT version used by 2635 and then run the tests - they should all pass. @t92549 - could you take a look at testing and confirming this locally?

**Review by t92549 — APPROVED** [review]

**Pull request merged**